src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_ioctl.c

```diff
@@ -540,6 +536,40 @@ xocl_read_axlf_helper(struct xocl_dev *xdev, struct drm_xocl_axlf *axlf_ptr)
 		goto done;
 	}
 
+	/* Populating MEM_TOPOLOGY sections. */
+	size = xocl_read_sect(MEM_TOPOLOGY, &new_topology, axlf, buf);
+	if (size <= 0) {
+		if (size != 0)
+			goto done;
```

Comment:
But you added memcmp(new_topology, xdev->topology, size). The old code makes sense, but does it work for your memcmp()? If size == 0, will memcmp() thinks the new topology is the same or different? Does the result match your expectation? Also, you've moved the code around, what makes sense before may not make sense any more. I don't understand why you think it's OK to leave it as is only because it was written that way at a different place?